ctsit/nacculator
# Fix FTLD follow-up Z1X form columns

[data_ftld.csv](https://github.com/ctsit/nacculator/files/7436736/data_ftld.csv)
**What does this change do?**

This change implements feedback from Anna Donnelly at Massachusetts General Hospital about FTLD issues her center was experiencing. I have corrected the column spacing in the FTLD follow-up visit Z1X form (NACC's Data Element Dictionary, that I based the code on, had the error and it carried over into NACCulator). I verified that Follow-up FTLD packets are now able to be submitted to NACC's system without column errors.

I also took the opportunity to correct some minor typographical errors and logging bugs while I was trying to re-create Ms. Donnelly's issue.


## Verification

_List the steps needed to make sure this thing works. Be precise._

1. Download Sam's Test FTLD Data (attached to this pull request at the top)
2. From the command-line, run:
redcap2nacc -ftld -ivp </path/to/data_ftld.csv >test.txt
for initial, or
redcap2nacc -ftld -fvp </path/to/data_ftld.csv >test.txt
for follow-up
3. The output should look like: a range error or invalid character error for test PTID 1, and normal .txt output for test PTID 2.
4. (relating to the error logging) If you use the "-tfp" flag instead of "-ivp -ftld", the output should look like this: 
Could not find a REDCap field for TFP Z1X form.
So that NACCulator alerts the user if it cannot find any of its expected fields for the Z1X in a telephone REDCap event.

- Verify the thing does this: works as NACCulator should and produces output files that can be submitted directly to NACC's system.
- Verify the thing does not do that: prints data in columns incompatible with NACC's submission system.


## Affirmations

All of these should have a check by them. Any exception requires an explanation.
